DSharpPlus/Entities/Channel/Message/DiscordMessage.cs

```diff
@@ -300,6 +300,12 @@ public IReadOnlyList<DiscordMessageSticker>? Stickers
     [JsonProperty("referenced_message", NullValueHandling = NullValueHandling.Ignore)]
     public DiscordMessage? ReferencedMessage { get; internal set; }
 
+    /// <summary>
+    /// Gets the message object for the referenced message
+    /// </summary>
+    [JsonProperty("message_snapshots", NullValueHandling = NullValueHandling.Ignore)]
+    public DiscordMessageSnapshot[]? MessageSnapshots { get; internal set; }

```

Comment:
This should be IReadOnlyList<DiscordMessageSnapshot> to align with other public APIs